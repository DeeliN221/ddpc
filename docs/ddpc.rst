ddpc package
============

Subpackages
-----------

.. toctree::
   :maxdepth: 4

   ddpc.io

Submodules
----------

ddpc.util module
----------------

.. automodule:: ddpc.util
   :members:
   :undoc-members:
   :show-inheritance:

Module contents
---------------

.. automodule:: ddpc
   :members:
   :undoc-members:
   :show-inheritance:
