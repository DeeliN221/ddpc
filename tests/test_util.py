"""Test main function in ddpc.util.find_prim."""

from pathlib import Path
from ddpc.util import find_prim, scale_atom_pos, find_orth


s_folder = Path(f"{__file__}/../structures").resolve()
(s_folder / "output").mkdir(exist_ok=True, parents=True)


def test_find_prim(snapshot):
    """Follow snakemake inp, out, par convention."""
    for s in s_folder.iterdir():
        # xyz has no lattice info
        if s.name.endswith(".as"):
            p = s
            op = s.parent / "output" / s.with_suffix(".prim").name
            fmt = "vasp"
            symprec = 1e-5

            find_prim(p, op, fmt, symprec)
            assert f"{s=}\n{op=}" == snapshot


def test_scale_atom_pos(snapshot):
    """Follow snakemake inp, out, par convention."""
    for s in s_folder.iterdir():
        # xyz has no lattice info
        if s.name.endswith(".as"):
            p = s
            op = s.parent / "output" / s.with_suffix(".POSCAR").name
            scale_atom_pos(p, op)
            assert f"{s=}\n{op=}" == snapshot


def test_find_orth(snapshot):
    """Follow snakemake inp, out, par convention."""
    for s in s_folder.iterdir():
        # xyz has no lattice info
        if s.name.endswith(".as"):
            p = s
            op = s.parent / "output" / s.with_suffix(".orth").name
            fmt = "vasp"
            mlen = 15.0

            find_orth(p, op, fmt, mlen)
            assert f"{s=}\n{op=}" == snapshot
