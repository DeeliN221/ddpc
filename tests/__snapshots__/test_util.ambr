# serializer version: 1
# name: test_find_orth
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/mag.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/mag.orth')
  '''
# ---
# name: test_find_orth.1
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/magxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/magxyz.orth')
  '''
# ---
# name: test_find_orth.2
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/fixxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/fixxyz.orth')
  '''
# ---
# name: test_find_orth.3
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/lat.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/lat.orth')
  '''
# ---
# name: test_find_orth.4
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/all.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/all.orth')
  '''
# ---
# name: test_find_orth.5
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/fixxyzmagxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/fixxyzmagxyz.orth')
  '''
# ---
# name: test_find_prim
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/mag.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/mag.prim')
  '''
# ---
# name: test_find_prim.1
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/magxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/magxyz.prim')
  '''
# ---
# name: test_find_prim.2
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/fixxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/fixxyz.prim')
  '''
# ---
# name: test_find_prim.3
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/lat.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/lat.prim')
  '''
# ---
# name: test_find_prim.4
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/all.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/all.prim')
  '''
# ---
# name: test_find_prim.5
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/fixxyzmagxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/fixxyzmagxyz.prim')
  '''
# ---
# name: test_scale_atom_pos
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/mag.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/mag.POSCAR')
  '''
# ---
# name: test_scale_atom_pos.1
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/magxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/magxyz.POSCAR')
  '''
# ---
# name: test_scale_atom_pos.2
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/fixxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/fixxyz.POSCAR')
  '''
# ---
# name: test_scale_atom_pos.3
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/lat.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/lat.POSCAR')
  '''
# ---
# name: test_scale_atom_pos.4
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/all.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/all.POSCAR')
  '''
# ---
# name: test_scale_atom_pos.5
  '''
  s=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/fixxyzmagxyz.as')
  op=PosixPath('/home/zzl/lab/mob_proj/ddpc/tests/structures/output/fixxyzmagxyz.POSCAR')
  '''
# ---
